### PR TITLE
clippy::perf fixes

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs [hunk 1]
```diff
@@ -222,8 +222,8 @@ impl ItemLikeVisitor<'v> for InherentOverlapChecker<'tcx> {
                                 let id_to_set = *ids.iter().min().unwrap();
 
                                 // Sort the id list so that the algorithm is deterministic
-                                let mut ids = ids.into_iter().collect::<SmallVec<[_; 8]>>();
-                                ids.sort();
+                                let mut ids = ids.into_iter().collect::<SmallVec<[usize; 8]>>();
+                                ids.sort_unstable();
```
Comment (klensy) [inline compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs:226]: Perhaps this (and lower) `sort`'s need to be stable?

Comment (matthiaskrgr) [reply compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs:226]: We are sorting primitives (`usize`s) here, so it does not matter if we have `1, 1, 1` or `1, 1, 1` :)

**Comment by jackh726** [comment]
@bors r+

Simple changes. Theoretical perf improvements, even if nothing on benches.

**Pull request merged**